buffer.c

```diff
@@ -454,25 +365,20 @@ zval *buffer_view_offset_get(buffer_view_object *intern, size_t offset)
 			/* Should never happen */
 			zend_error_noreturn(E_ERROR, "Invalid buffer view type");
 	}
-
-	return retval;
 }
 
 void buffer_view_offset_set(buffer_view_object *intern, long offset, zval *value)
 {
 	if (intern->type == buffer_view_float || intern->type == buffer_view_double) {
-		Z_ADDREF_P(value);
-		convert_to_double_ex(&value);
-
 		if (intern->type == buffer_view_float) {
-			intern->buf.as_float[offset] = Z_DVAL_P(value);
+			intern->buf.as_float[offset] = zval_get_double(value);
 		} else {
-			intern->buf.as_double[offset] = Z_DVAL_P(value);
+			intern->buf.as_double[offset] = zval_get_double(value);
 		}
 
-		zval_ptr_dtor(&value);
+		zval_ptr_dtor(value);
```

Comment:
This probably needs to be removed, this is now destroying the offset argument, not the conversion result.